Variable set but not used compile errors on macOS 12.2.1 and XCode 13.3

Received the following errors when following the [JDK 11 compilation steps on macOS](https://github.com/eclipse-openj9/openj9/blob/master/doc/build-instructions/Build_Instructions_V11.md#macOS):

```
/Users/kevin/git/test/omr/port/osx/omrosdump.c:343:15: error: variable 'data_read' set but not used [-Werror,-Wunused-but-set-variable]
/Users/kevin/git/test/omr/port/osx/omrosdump.c:344:26: error: variable 'data_size' set but not used [-Werror,-Wunused-but-set-variable]
/Users/kevin/git/test/omr/gc/base/standard/CompactScheme.cpp:1199:12: error: variable 'evacuatedBytes' set but not used [-Werror,-Wunused-but-set-variable]
/Users/kevin/git/test/omr/gc/base/standard/ConcurrentCardTable.cpp:1048:12: error: variable 'objects' set but not used [-Werror,-Wunused-but-set-variable]
```

macOS and XCode versions:

```
$ system_profiler SPSoftwareDataType SPDeveloperToolsDataType
Software:

    System Software Overview:

      System Version: macOS 12.2.1 (21D62)
      Kernel Version: Darwin 21.3.0
      Boot Volume: MainDisk
      Boot Mode: Normal
      Computer Name: Kevin's MacBook Pro
      User Name: Kevin Grigorenko (kevin)
      Secure Virtual Memory: Enabled
      System Integrity Protection: Disabled
      Time since boot: 16 days 2:27

Developer:

    Developer Tools:

      Version: 13.3 (13E113)
      Location: /Applications/Xcode.app
      Applications:
          Xcode: 13.3 (20102)
          Instruments: 13.3 (64552.70)
```

--- a/gc/base/standard/CompactScheme.cpp
+++ b/gc/base/standard/CompactScheme.cpp
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 1991, 2021 IBM Corp. and others
+ * Copyright (c) 1991, 2022 IBM Corp. and others
  *
  * This program and the accompanying materials are made available under
  * the terms of the Eclipse Public License 2.0 which accompanies this

--- a/gc/base/standard/CompactScheme.cpp
+++ b/gc/base/standard/CompactScheme.cpp
@@ -1196,7 +1196,6 @@ MM_CompactScheme::doCompact(MM_EnvironmentStandard *env, MM_MemorySubSpace *memo
 
 	omrobjectptr_t objectPtr = 0;
 	omrobjectptr_t nextObject = 0;
-	uintptr_t evacuatedBytes = 0;
 	intptr_t page = -1; /* invalid value */
 	intptr_t counter = 0; /* obj on page, first is zero */
 	CompactTableEntry entry;
@@ -1220,7 +1219,6 @@ MM_CompactScheme::doCompact(MM_EnvironmentStandard *env, MM_MemorySubSpace *memo
 			if (pageByteCount > deadObjectSize) {
 				break;
 			}
-			evacuatedBytes += pageByteCount;
 		}
 
 		uintptr_t objectSize = _extensions->objectModel.getConsumedSizeInBytesWithHeader(objectPtr);

--- a/gc/base/standard/ConcurrentCardTable.cpp
+++ b/gc/base/standard/ConcurrentCardTable.cpp
@@ -1045,7 +1045,6 @@ MM_ConcurrentCardTable::finalCleanCards(MM_EnvironmentBase *env, uintptr_t *byte
 	uintptr_t traceCount = 0;
 	Card * nextDirtyCard;
 	omrobjectptr_t objectPtr;
-	uintptr_t objects;
 	uintptr_t cards = 0;
 	bool phase2 = false;
 
@@ -1082,9 +1081,7 @@ MM_ConcurrentCardTable::finalCleanCards(MM_EnvironmentBase *env, uintptr_t *byte
 
 		/* Then iterate over all marked objects in the heap between the two addresses */
 		MM_HeapMapIterator markedObjectIterator(_extensions, markMap, heapBase, heapTop);
-		objects = 0;
 		while (NULL != (objectPtr = markedObjectIterator.nextObject())) {
-			objects +=1;
 			traceCount += _markingScheme->scanObject(env, objectPtr, SCAN_REASON_DIRTY_CARD);
 		}
 

--- a/gc/base/standard/ConcurrentCardTable.cpp
+++ b/gc/base/standard/ConcurrentCardTable.cpp
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 1991, 2017 IBM Corp. and others
+ * Copyright (c) 1991, 2022 IBM Corp. and others
  *
  * This program and the accompanying materials are made available under
  * the terms of the Eclipse Public License 2.0 which accompanies this

--- a/port/osx/omrosdump.c
+++ b/port/osx/omrosdump.c
@@ -340,8 +340,6 @@ list_segment_commands(mach_port_t task_port, struct segment_command_64 **segment
 		vm_size_t size = 0;
 		vm_address_t old_addr = address;
 		mach_msg_type_number_t info_count = VM_REGION_SUBMAP_INFO_COUNT_64;
-		vm_offset_t data_read = 0;
-		mach_msg_type_number_t data_size = 0;
 
 		if (*segment_count >= segment_array_size) {
 			segment_array_size += segment_array_size / 2;
@@ -385,9 +383,6 @@ list_segment_commands(mach_port_t task_port, struct segment_command_64 **segment
 		segments[*segment_count].maxprot    = vm_info.max_protection;
 		segments[*segment_count].initprot   = vm_info.protection;
 		segments[*segment_count].nsects     = 0;
-
-		data_read = 0;
-		data_size = 0;
 
 		*segment_count += 1;
 		address += size;

